### ISSUE TITLE
vsmac 7.8 b1468 doesn't start with mono 5.16

### ISSUE THREAD

**Issue by StephaneDelcroix** [mono/monodevelop#6779]
updated the packages from the updater this morning, and vsmac fails at startup

```
/Applications/Visual\ Studio.app/Contents/MacOS/VisualStudio --no-redirect
INFO [2018-12-17 14:05:44Z]: Starting Visual Studio 2017 for Mac (Preview) 7.8 Preview (7.8 build 1468)
INFO [2018-12-17 14:05:44Z]: Build Information
Release ID: 708001468
Git revision: 4d7cc66b885f00552132078a86b7e43adddf8ed2
Build date: 2018-12-14 18:56:30+00
Build branch: release-7.8
Xamarin extensions: dccbb397b2724d3cd880ed864f627b812640952f

INFO [2018-12-17 14:05:44Z]: Running on Mono 5.16.0.226 (2018-06/ad3edd34840) (64-bit)
INFO [2018-12-17 14:05:44Z]: Operating System: Mac OS X 10.14.3
Darwin 18.2.0 Darwin Kernel Version 18.2.0
    Tue Dec  4 17:09:09 PST 2018
    root:xnu-4903.240.8~10/RELEASE_X86_64 x86_64
INFO [2018-12-17 14:05:44Z]: Initializing ms error reporting app: com.microsoft.visual-studio, Visual Studio, 7.8.0.1468
INFO [2018-12-17 14:05:44Z]:                                    : /Applications/Visual Studio.app
INFO [2018-12-17 14:05:44Z]: GTK: Using gtkrc from /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/gtkrc.mac
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) object.wrapper_native_0x102f47470 () [0x00008] in <26329eb189424f5cbc71e5fa393e447e>:0
  at ObjCRuntime.Runtime.EnsureInitialized () [0x00041] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.1/src/Xamarin.Mac/ObjCRuntime/Runtime.mac.cs:118
  at AppKit.NSApplication.Init () [0x00016] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.1/src/Xamarin.Mac/AppKit/NSApplication.cs:56
  at Xwt.Mac.NSApplicationInitializer.Initialize () [0x00046] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt.XamMac/Xwt.Mac/NSApplicationInitializer.cs:44
  at Xwt.Mac.MacEngine.InitializeApplication () [0x00000] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt.XamMac/Xwt.Mac/MacEngine.cs:48
  at Xwt.Backends.ToolkitEngineBackend.Initialize (Xwt.Toolkit,bool) [0x00032] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt.Backends/ToolkitEngineBackend.cs:58
  at Xwt.Toolkit.Initialize (bool) [0x00016] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Toolkit.cs:305
  at Xwt.Toolkit.LoadBackend (string,bool,bool) [0x00059] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Toolkit.cs:288
  at Xwt.Toolkit.Load (Xwt.ToolkitType) [0x00040] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Toolkit.cs:235
  at MonoDevelop.Components.IdeTheme.InitializeGtk (string,string[]&) [0x00049] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/IdeTheme.cs:76
  at MonoDevelop.Ide.IdeStartup.Run (MonoDevelop.Ide.MonoDevelopOptions) [0x001a2] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:142
  at MonoDevelop.Ide.IdeStartup.Main (string[],MonoDevelop.Ide.Extensions.IdeCustomizer) [0x000bc] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:843
  at Xamarin.Startup.MainClass.Main (string[]) [0x00000] in /Users/vsts/agent/2.144.0/work/1/s/md-addins/Xamarin.Startup/Main.cs:11
  at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) [0x00057] in <627a8c3ca3794ecc8b5cfe36628de233>:0
FATAL ERROR [2018-12-17 14:05:44Z]: Visual Studio failed to start. Some of the assemblies required to run Visual Studio (for example gtk-sharp)may not be properly installed in the GAC.
System.Exception: Toolkit could not be loaded ---> System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper managed-to-native) System.Object.wrapper_native_0x102f47470()
  at ObjCRuntime.Runtime.EnsureInitialized () [0x00041] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.1/src/Xamarin.Mac/ObjCRuntime/Runtime.mac.cs:118 
  at AppKit.NSApplication.Init () [0x00016] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.1/src/Xamarin.Mac/AppKit/NSApplication.cs:56 
  at Xwt.Mac.NSApplicationInitializer.Initialize () [0x00046] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt.XamMac/Xwt.Mac/NSApplicationInitializer.cs:44 
  at Xwt.Mac.MacEngine.InitializeApplication () [0x00000] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt.XamMac/Xwt.Mac/MacEngine.cs:48 
  at Xwt.Backends.ToolkitEngineBackend.Initialize (Xwt.Toolkit toolkit, System.Boolean isGuest) [0x00032] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt.Backends/ToolkitEngineBackend.cs:58 
  at Xwt.Toolkit.Initialize (System.Boolean isGuest) [0x00016] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Toolkit.cs:305 
  at Xwt.Toolkit.LoadBackend (System.String type, System.Boolean isGuest, System.Boolean throwIfFails) [0x00059] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Toolkit.cs:288 
   --- End of inner exception stack trace ---
  at Xwt.Toolkit.LoadBackend (System.String type, System.Boolean isGuest, System.Boolean throwIfFails) [0x0006c] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Toolkit.cs:295 
  at Xwt.Toolkit.Load (Xwt.ToolkitType type) [0x00040] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Toolkit.cs:235 
  at MonoDevelop.Components.IdeTheme.InitializeGtk (System.String progname, System.String[]& args) [0x00049] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/IdeTheme.cs:76 
  at MonoDevelop.Ide.IdeStartup.Run (MonoDevelop.Ide.MonoDevelopOptions options) [0x001a2] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:142 
  at MonoDevelop.Ide.IdeStartup.Main (System.String[] args, MonoDevelop.Ide.Extensions.IdeCustomizer customizer) [0x000bc] in /Users/vsts/agent/2.144.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:843 
```

> VS bug [#752814](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/752814)

**Comment by StephaneDelcroix** [comment]
I got the same issue with vsmac 8

**Comment by Therzok** [comment]
Looks more like a xammac/mono incompatibility rather than vsmac issue.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-21] Comment by **xamarinc**



Fixed in version 7.8.0.1547 (release-7.8)



Pull Request #4046 merged by: Lluis Sanchez  
 Author: xamarin  
 Commit: becb77097be5b754a310dbfe39dc9a326e73620e (xamarin/md-addins)  
 Included in Commit: ac9638e6096d900f180fc470f148ea8fa059ee42 (mono/monodevelop)
